Stripe/STPAddressFieldTableViewCell.swift

```diff
@@ -0,0 +1,505 @@
+      self.delegate?.availableCountries ?? Set(NSLocale.isoCountryCodes))
+    if otherCountryCodes.contains(countryCode ?? "") {
+      // Remove the current country code to re-add it once we sort the list.
+      otherCountryCodes.removeAll { $0 == countryCode }
+    } else {
+      // If it isn't in the list (if we've been configured to not show that country), don't re-add it.
+      countryCode = nil
+    }
+    let locale = NSLocale.current as NSLocale
+    otherCountryCodes =
+      (otherCountryCodes as NSArray).sortedArray(comparator: { code1, code2 in
+        guard let code1 = code1 as? String, let code2 = code2 as? String else {
+          return .orderedDescending
+        }
+        let localeID1 = NSLocale.localeIdentifier(fromComponents: [
+          NSLocale.Key.countryCode.rawValue: code1
+        ])
+        let localeID2 = NSLocale.localeIdentifier(fromComponents: [
+          NSLocale.Key.countryCode.rawValue: code2
+        ])
+        let name1 = locale.displayName(forKey: .identifier, value: localeID1)
+        let name2 = locale.displayName(forKey: .identifier, value: localeID2)
+        return (name1?.compare(name2 ?? ""))!
+      }) as? [String] ?? []
+    if let countryCode = countryCode {
+      countryCodes = ["", countryCode] + otherCountryCodes
+    } else {
+      countryCodes = [""] + otherCountryCodes
+    }
+    let pickerView = UIPickerView()
+    pickerView.dataSource = self
+    pickerView.delegate = self
+    countryPickerView = pickerView
+
+    self.lastInList = lastInList
+    self.type = type
+    self.textField?.text = contents
+
+    var ourCountryCode = self.delegate?.addressFieldTableViewCountryCode
+
+    if ourCountryCode == nil {
+      ourCountryCode = countryCode
+    }
+    delegateCountryCodeDidChange(countryCode: ourCountryCode ?? "")
+    updateAppearance()
+
+    self.textField?.translatesAutoresizingMaskIntoConstraints = false
+
+    if let textField = self.textField {
+      NSLayoutConstraint.activate(
+        [
+          textField.leadingAnchor.constraint(
+            equalTo: contentView.safeAreaLayoutGuide.leadingAnchor, constant: 15),
+          textField.trailingAnchor.constraint(
+            equalTo: contentView.safeAreaLayoutGuide.trailingAnchor, constant: -15),
+          textField.topAnchor.constraint(
+            equalTo: contentView.safeAreaLayoutGuide.topAnchor, constant: 1),
+          contentView.safeAreaLayoutGuide.bottomAnchor.constraint(
+            greaterThanOrEqualTo: textField.bottomAnchor),
+          textField.heightAnchor.constraint(greaterThanOrEqualToConstant: 43),
+          inputAccessoryToolbar?.heightAnchor.constraint(equalToConstant: 44),
+        ].compactMap { $0 })
+    }
+  }
+
+  var type: STPAddressFieldType = .name
+  var caption: String? {
+    get {
+      self.textField?.placeholder
+    }
+    set {
+      self.textField?.placeholder = newValue
+    }
+  }
+
+  private(set) weak var textField: STPValidatedTextField?
```

Comment:
I don't think this should be optional

Reply:
This was the case in Objective-C, and a strong reference would be redundant, but I don't see a good reason not to have one. I'll change this